### ISSUE TITLE
Absolute date at HTTP API query parameter "from" and "until" doesn't work properly when today is February 29th

### ISSUE THREAD

**Issue by arifiswise** [graphite-project/graphite-web#1477]
FOUND IN: graphite v0.9.10
USE CASE:
Suppose that today is February 29th 2016.
Try to call HTTP API to retrieve last year (2015) statistics by using query parameter "from" and "until".
Following error will be thrown:

```
Traceback (most recent call last):
  File "/opt/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/opt/graphite/webapp/graphite/render/views.py", line 45, in renderView
    (graphOptions, requestOptions) = parseOptions(request)
  File "/opt/graphite/webapp/graphite/render/views.py", line 239, in parseOptions
    fromTime = parseATTime( queryParams['from'] )
  File "/opt/graphite/webapp/graphite/render/attime.py", line 45, in parseATTime
    return tzinfo.localize(parseTimeReference(ref), daylight) + parseTimeOffset(offset)
  File "/opt/graphite/webapp/graphite/render/attime.py", line 94, in parseTimeReference
    refDate = refDate.replace(year= int(ref[:4]))
ValueError: day is out of range for month
```

This issue can be fixed by modifying 
/opt/graphite/webapp/graphite/render/attime.py, line 94: at following part:

```
  elif len(ref) == 8 and ref.isdigit(): #YYYYMMDD
    refDate = refDate.replace(year= int(ref[:4]))

    try: # Fix for Bug #551771
      refDate = refDate.replace(month= int(ref[4:6]))
      refDate = refDate.replace(day= int(ref[6:8]))
    except:
      refDate = refDate.replace(day= int(ref[6:8]))
      refDate = refDate.replace(month= int(ref[4:6]))
```

become:

```
  elif len(ref) == 8 and ref.isdigit(): #YYYYMMDD
    refDate = refDate.replace(day=1)
    refDate = refDate.replace(year= int(ref[:4]))

    try: # Fix for Bug #551771
      refDate = refDate.replace(month= int(ref[4:6]))
      refDate = refDate.replace(day= int(ref[6:8]))
    except:
      refDate = refDate.replace(day= int(ref[6:8]))
      refDate = refDate.replace(month= int(ref[4:6]))
```

**Comment by deniszh** [comment]
Yep, looks like need to be fixed in 0.9.x and master too

**Comment by bmhatfield** [comment]
I refactored a few things in 0.9.14/15 w/r/t attime - this may already be fixed.
